### PR TITLE
111 house details

### PR DESCRIPTION
**Pull request by mAbadsa**
Get user data to show it in more details page

### CODE REVIEW

**Review by mariamisa — CHANGES_REQUESTED** [review]
I told you to send it through props, why u make it like that !! 
you can make a nested route and send the data

File: client/src/Components/Card/index.js [hunk 1]
```diff
@@ -80,7 +85,27 @@ export default function CardComponent({
         </CardContent>
       </CardActionArea>
       <CardActions className={classes.cardAction}>
-        <Button className={classes.border}>more details</Button>
+        <Link
+          to={{
+            pathname: `/houses/${id}`,
```
Comment (okjake) [inline client/src/Components/Card/index.js:90]: You could use your path constants to create this pathname to make sure they stay in sync

File: client/src/Pages/DetailsHouse/index.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      price,
+      username,
+      email,
+      mobile,
+    },
+  } = useLocation();
+
+  useEffect(() => {
+    window.scrollTo(0, 0);
+  }, []);
```
Comment (mariamisa) [inline client/src/Pages/DetailsHouse/index.js:33]: delete  use effect

File: client/src/Pages/DetailsHouse/index.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+  const getCityName = (locationId) => {
+    const { city } = locations.find(({ id: _id }) => _id === locationId);
+    return city;
+  };
```
Comment (mariamisa) [inline client/src/Pages/DetailsHouse/index.js:38]: I like the way you find the data :+1:

File: client/src/Pages/DetailsHouse/index.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            </Button>
+            <Link to={`${HOME_PAGE}`} className={classes.backLink}>
+              Back
+            </Link>
+          </div>
+        </Grid>
+      </Grid>
+      <Grid lg="12">
+        <RelatedHouse />
+      </Grid>
```
Comment (mariamisa) [inline client/src/Pages/DetailsHouse/index.js:88]: for now, delete the related houses (related folder)

File: client/src/Pages/DetailsHouse/RelatedHouse/index.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+      availability: true,
+      room_num: 3,
+      bathroom_num: 2,
+      price: '90.00',
+      area: '100.00',
+      img:
+        'https://us.123rf.com/450wm/iriana88w/iriana88w1711/iriana88w171100467/89727643-beautiful-exterior-of-contemporary-home-with-two-car-garage-spaces-at-sunset-northwest-usa.jpg?ver=6',
+      rating: 9,
+      created_at: '2021-02-06T09:07:27.315Z',
+    },
```
Comment (mariamisa) [inline client/src/Pages/DetailsHouse/RelatedHouse/index.js:35]: there is no need to use fake data, delete this file

File: client/src/Components/Card/index.js [hunk 6]
```diff
@@ -80,7 +86,27 @@ export default function CardComponent({
...
+              img,
+              title,
+              description,
+              location,
+              bdCount,
+              baCount,
+              price,
+              username,
+              email,
+              mobile,
```
Comment (mariamisa) [inline client/src/Components/Card/index.js:103]: inside the card component display owner name and mobile

File: client/src/Pages/DetailsHouse/index.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+      img,
+      title,
+      description,
+      location,
+      price,
+      username,
+      email,
+      mobile,
+    },
+  } = useLocation();
```
Comment (mariamisa) [inline client/src/Pages/DetailsHouse/index.js:29]: what will happen if we type the URL (/houses/2), how can handle this problem?
* hint ** you can access to id with useParams

**Review by mariamisa — CHANGES_REQUESTED** [review]

File: server/router/publicRoutes.js [hunk 8]
```diff
@@ -23,6 +24,7 @@ router.post('/login', loginValidation, loginController);
 
 router.get('/houses', getAllHouses);
 router.get('/newest-houses', getNewestHouses);
+router.get('/houses/id/:houseId', getHouseById);
```
Comment (mariamisa) [inline server/router/publicRoutes.js:27]: fix endpoint name to ``` house/:houseId ```

File: server/database/queries/house/getNewestHousesData.js [hunk 9]
```diff
@@ -3,7 +3,7 @@ const connection = require('../../config/connection');
 const getNewestHousesdata = (limit = 10, skip = 0) => {
   const sql = {
     text:
-      'SELECT * FROM (SELECT * FROM houses ORDER BY id DESC LIMIT $1 OFFSET $2) newest ORDER BY id ASC;',
+      'SELECT nh.*, u.username, u.email, u.mobile FROM (SELECT * FROM houses ORDER BY id DESC LIMIT $1 OFFSET $2) AS nh INNER JOIN users AS u ON nh.user_id = u.id ORDER BY id ASC;',
```
Comment (mariamisa) [inline server/database/queries/house/getNewestHousesData.js:6]: this query doesn't work as expected (make a test)

**Review by mariamisa — CHANGES_REQUESTED** [review]
why you write the backend code with the front end !!

**Review by yasmeenattallah — DISMISSED** [review]

**Review by yasmeenattallah — APPROVED** [review]

File: server/database/queries/house/getNewestHousesData.js (hunk 9 above)
Comment (mAbadsa) [reply server/database/queries/house/getNewestHousesData.js]: It works fine.

**Review by mariamisa — APPROVED** [review]

**Pull request merged**